File: app/src/main/java/com/saitejajanjirala/imdbclone/db/DatabaseService.java
package com.saitejajanjirala.imdbclone.db;

import androidx.room.Database;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;


import com.saitejajanjirala.imdbclone.db.dao.ResultDao;

import com.saitejajanjirala.imdbclone.models.Popular;
import com.saitejajanjirala.imdbclone.models.Result;

import org.jetbrains.annotations.NotNull;

@Database(
        entities = { Result.class},
        exportSchema = false,
        version = 1
)
@TypeConverters({IdsToStringConverter.class})
public abstract class DatabaseService extends RoomDatabase {

    @NotNull
    public abstract ResultDao resultDao();
}

File: app/src/main/java/com/saitejajanjirala/imdbclonekotlin/db/DatabaseService.kt
package com.saitejajanjirala.imdbclonekotlin.db

import androidx.room.Database
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import com.saitejajanjirala.imdbclonekotlin.db.dao.ResultDao
import com.saitejajanjirala.imdbclonekotlin.models.Result

@Database(entities = [Result::class], exportSchema = false, version = 1)
@TypeConverters(
    IdsToStringConverter::class
)
abstract class DatabaseService : RoomDatabase() {
    abstract fun resultDao(): ResultDao
}